sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,222 @@
+        the same time as ``func`` and ``inverse_func``. If ``None`` and
+        ``func`` and ``inverse_func`` are ``None`` as well, the transformer
+        will be an identity transformer.
+
+    func : function, (default=None)
+        Function to apply to ``y`` before passing to ``fit``. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` is
+        ``None`` as well, the function used will be the identity function.
+
+    inverse_func : function, (default=None)
+        Function apply to the prediction of the estimator. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` as
+        well, the function used will be the identity function.
+
+    check_invertible : bool, (default=True)
+        Whether to check that ``transform`` followed by ``inverse_transform``
+        or ``func`` followed by ``inverse_func`` lead to the original data.
+
+    Attributes
+    ----------
+    estimator_ : object
+        Fitted estimator.
+
+    transformer_ : object
+        Used transformer in ``fit`` and ``predict``.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.preprocessing.label import TransformedTargetRegressor
+    >>> tt = TransformedTargetRegressor(estimator=LinearRegression(),
+    ...                                 func=np.log, inverse_func=np.exp)
+    >>> X = np.arange(4).reshape(-1, 1)
+    >>> y = np.exp(2 * X).ravel()
+    >>> tt.fit(X, y)
+    ... #doctest: +NORMALIZE_WHITESPACE
+    TransformedTargetRegressor(check_invertible=True,
+        estimator=LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1,
+                                   normalize=False),
+        func=<ufunc 'log'>, inverse_func=<ufunc 'exp'>, transformer=None)
+    >>> tt.score(X, y)
+    1.0
+    >>> tt.estimator_.coef_
+    array([ 2.])
+
+    """
+    def __init__(self, estimator=LinearRegression(), transformer=None,
+                 func=None, inverse_func=None, check_invertible=True):
+        self.estimator = estimator
+        self.transformer = transformer
+        self.func = func
+        self.inverse_func = inverse_func
+        self.check_invertible = check_invertible
+        # we probably need to change this ones we have tags
+        self._estimator_type = estimator._estimator_type
+
+    def _validate_transformer(self, y):
+        if (self.transformer is not None and
+                (self.func is not None or self.inverse_func is not None)):
+            raise ValueError("Both 'transformer' and functions 'func'/"
+                             "'inverse_func' cannot be set at the same time.")
+        elif self.transformer is not None:
+            self.transformer_ = clone(self.transformer)
+        else:
+            self.transformer_ = FunctionTransformer(
+                func=self.func, inverse_func=self.inverse_func, validate=False)
+        self.transformer_.fit(y)
```

Comment:
Do you mean that I should call `fit_transform` for `check_inverse` and as well at `fit` time?